lib/helpers/metaLoader.js

```diff
@@ -33,81 +32,85 @@ const TYPES_MAP = {
+  }
+
+  async ObjectMetaData() {
+    const objMetaData = await this.getObjectMetaData();
+
+    return {
+      findFieldByLabel: function findFieldByLabel(fieldLabel) {
+        return objMetaData.fields.find(field => field.label === fieldLabel);
+      },
+      isStringField: function isStringField(field) {
+        return field && (field.soapType === 'tns:ID' || field.soapType === 'xsd:string');
+      },
+    };
   }
 
   async getLookupFieldsModel() {
-    return await this.connection.describe(this.configuration.sobject)
-      .then(async meta => {
+    return this.connection.describe(this.configuration.sobject)
+      .then(async (meta) => {
         const model = {};
         await meta.fields
-          .filter(
-            field => field.externalId || field.unique || field.name ===
-              'Id' || field.type === 'reference')
-          .forEach(field => {
+          .filter(field => field.externalId || field.unique || field.name === 'Id' || field.type === 'reference')
+          .forEach((field) => {
             model[field.name] = field.label;
           });
         return model;
       });
   }
 
   async getLookupFieldsModelWithTypeOfSearch(typeOfSearch) {
-    if (typeOfSearch === 'uniqueFields')
-    return await this.connection.describe(this.configuration.sobject)
-      .then(async meta => {
+    if (typeOfSearch === 'uniqueFields') {
+      return this.connection.describe(this.configuration.sobject)
+        .then(async (meta) => {
+          const model = {};
+          await meta.fields
+            .filter(field => field.type === 'id' || field.unique)
+            .forEach((field) => {
+              model[field.name] = `${field.label} (${field.name})`;
+            });
+          return model;
+        });
+    }
+    return this.connection.describe(this.configuration.sobject)
+      .then(async (meta) => {
         const model = {};
         await meta.fields
-          .filter(field => field.type === 'id' || field.unique)
-          .forEach(field => {
+          .forEach((field) => {
             model[field.name] = `${field.label} (${field.name})`;
           });
         return model;
       });
-    else
-      return await this.connection.describe(this.configuration.sobject)
-          .then(async meta => {
-            const model = {};
-            await meta.fields
-                .forEach(field => {
-                  model[field.name] = `${field.label} (${field.name})`;
-                });
-            return model;
-          });
   }
 
   async loadMetadata() {
-    return await this.connection.describe(this.configuration.sobject)
-      .then(async meta => await this.processMeta(meta)).then(metaData => {
-        debug('emitting Metadata %j', metaData);
+    return this.connection.describe(this.configuration.sobject)
+      .then(async meta => this.processMeta(meta)).then((metaData) => {
+        this.emitter.logger.debug('emitting Metadata %j', metaData);
```

Comment:
You don't need to use `this.emitter.logger` in case of usage sailor version 2.5.1+, just replace with `this.logger`

Reply:
In this case `this` points on instance of MetaLoader class.
`this.emitter` contains `this` with logger.